apps/root/src/state/balances/actions.ts

```diff
@@ -65,10 +69,40 @@ export const fetchPricesForAllChains = createAppAsyncThunk<void, void>(
   }
 );
 
+const fetchTokenDetails = createAppAsyncThunk<
+  Token,
+  { tokenAddress: string; chainId: number; walletAddress: string; tokenList: TokenList }
+>(
+  'balances/fetchTokenDetails',
+  async ({ tokenAddress, chainId, walletAddress, tokenList }, { dispatch, extra: { web3Service } }) => {
+    if (tokenList[tokenAddress]) {
```

Comment:
is this already chainSpecific? bc we could be fetching a token from another chain if it has the same address

Reply:
This `tokenList` is from a specific chain. That is handled by `fetchInitialBalances` when it calls this function

Reply:
Perfect